application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,160 @@
+ * or Guild.
+ * <p>
+ * Actions are then logged to a dedicated channel, given by
+ * {@link Config#getModAuditLogChannelPattern()}.
+ */
+
+public class ModMailCommand extends SlashCommandAdapter {
+
+    private static final String COMMAND_NAME = "modmail";
+    private static final String MESSAGE = "message";
+    private static final String OPTION_MESSAGE_PRIVATE = "message-private";
+    private static final int COOLDOWN_DURATION_VALUE = 30;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private final Cache<Long, Instant> channelIdToLastCommandInvocation;
+    private final Predicate<String> modMailChannelNamePredicate;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config the config to use for this
+     */
+    public ModMailCommand(@NotNull Config config) {
+        super(COMMAND_NAME, "Sends a message to the moderators", SlashCommandVisibility.GLOBAL);
+        getData().addOption(OptionType.STRING, MESSAGE, "Message to the moderators", true)
+            .addOption(OptionType.BOOLEAN, OPTION_MESSAGE_PRIVATE,
+                    "do you wish for your message to stay private", false);
+
+        channelIdToLastCommandInvocation = setCooldown();
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    @NotNull
+    private Cache<Long, Instant> setCooldown() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandInteractionEvent event) {
+        MessageChannel messageChannel = event.getChannel();
+        if (isChannelOnCooldown(messageChannel)) {
+            event
+                .reply("Please wait a bit, this command can only be used once per %d %s.".formatted(
+                        COOLDOWN_DURATION_VALUE,
+                        COOLDOWN_DURATION_UNIT.toString().toLowerCase(Locale.US)))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        channelIdToLastCommandInvocation.put(messageChannel.getIdLong(), Instant.now());
+
+        Optional<TextChannel> auditLogChannel = getChannel(event);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        List<ModAuditLogWriter.Attachment> attachments = getAttachments(event);
+
+        String user = getName(event);
+        MessageAction message = auditLogChannel.orElseThrow().sendMessageEmbeds(messageEmbed(user));
+
+        message = buildAttachment(attachments, message);
+        message.queue();
+
+        event.reply("Thank you for contacting the moderators").setEphemeral(true).queue();
+
+    }
+
+    @NotNull
+    private String getName(@NotNull SlashCommandInteractionEvent event) {
+        String user = "@" + event.getUser().getName();
```

Comment:
https://github.com/Together-Java/TJ-Bot/pull/493#issuecomment-1214429489

_Yeah, that make sense. Would it be nicer UI to have message in the embed, and just have title with "Modmail from @user"?_

That is why I did it  ^

Reply:
I think he'll be fine with Tijs#0002 instead, As just the name isn't enough to identify someone

Reply:
Yeah, I don't mind the tag, it's much better than having just a name. I said `@user` because I think mention is preferable to just tag, if it is possible. Not really important, just a nicer UX.

So try `User#getAsMention`.

To quote myself from that issue:
> Would it be possible to ping the user that sent the modmail, instead of just saying 'user#123'. If user is pinged, we can just right-click and message them, or copy id, without searching for them. (User wouldn't actually be pinged, because they have no permissions to view the channel)

Reply:
It gives me a tag like this when I do it, if you guys are fine with it? `<@975000935698935858>`

Reply:
Yeah, that's how discord mentions work. Just make sure it renders properly. We want an actual clickable mention.

Something like this:
![image](https://user-images.githubusercontent.com/30927961/185705141-3e99f405-a5d4-4bab-8fb8-c1a74b5f366e.png)

Instead of this:
![image](https://user-images.githubusercontent.com/30927961/185705400-d0baecc0-59bd-4d8d-a2c7-3d7413c3d3fc.png)

Note that you don't have to do anything special for it to render properly, the string you pasted `<@975000935698935858>` works just fine. :relaxed:

Reply:
Done